lib/addons/jupyterhub/index.ts

```diff
@@ -67,64 +105,230 @@ export class JupyterHubAddOn extends HelmAddOn {
+                    "token_url": this.options.oidcConfig.tokenUrl,
+                    "userdata_url": this.options.oidcConfig.userDataUrl,
+                    scope:  this.options.oidcConfig.scope,
+                    username_key:  this.options.oidcConfig.usernameKey,
+                }
             });
-            setPath(values, "hub.config.JupyterHub.authenticator_class", "generic-oauth");
         }
 
         // Ingress instead of LoadBalancer service to expose the proxy - leverages AWS ALB
         // If not, then it will leverage AWS NLB
         const enableIngress = this.options.enableIngress || false;
+        const ingressHosts = this.options.ingressHosts || [];
+        const ingressAnnotations = this.options.ingressAnnotations;
         setPath(values, "ingress.enabled", enableIngress);
 
         if (enableIngress){
-            setPath(values, "ingress.annotations", 
-                {
-                    "kubernetes.io/ingress.class": "alb",
-                    "alb.ingress.kubernetes.io/scheme": "internet-facing",
-                    "alb.ingress.kubernetes.io/target-type": "ip",
-                }
-            );
-            setPath(values, "proxy.service.type", "NodePort");
+            setPath(values, "ingress.annotations", ingressAnnotations);
+            setPath(values, "ingress.hosts", ingressHosts);
+            setPath(values, "proxy.service", {"type" : "NodePort"});
         } else {
-            setPath(values, "proxy.service.annotations",
-                {
+            assert(!ingressHosts || ingressHosts.length == 0, 'Ingress Hosts CANNOT be assigned when ingress is disabled');
+            assert(!ingressAnnotations, 'Ingress annotations CANNOT be assigned when ingress is disabled');
+            setPath(values, "proxy.service", { 
+                "annotations": {
                     "service.beta.kubernetes.io/aws-load-balancer-type": "nlb",
                     "service.beta.kubernetes.io/aws-load-balancer-scheme": "internet-facing",
                     "service.beta.kubernetes.io/aws-load-balancer-nlb-target-type": "ip",
                 }
-            );
+            });
         }
 
-        // Create Namespace
-        const ns = createNamespace(this.options.namespace!, cluster, true, true);
-
         // Create Helm Chart
         const jupyterHubChart = this.addHelmChart(clusterInfo, values, false, true);
 
         // Add dependency
         jupyterHubChart.node.addDependency(ns);
         return Promise.resolve(jupyterHubChart);
     }
-}
\ No newline at end of file
+    /**
+     * This is a helper function to create EBS persistent storage
+     * @param {ClusterInfo} clusterInfo - Cluster Info
+     * @param {string} values - Helm Chart Values
+     * @param {string} ebsConfig - EBS Configurations supplied by user
+     * @returns
+     */
+    protected addEbsStorage(clusterInfo: ClusterInfo, values: any, ebsConfig: any){
+        const ebsAddon = 'EbsCsiDriverAddOn';
```

Comment:
minor: you don't need this constant. You can safely import the addon and use `getScheduledAddOn(EbsCsiDriverAddOn.name)`.
Example: https://github.com/aws-quickstart/cdk-eks-blueprints/blob/f51c11337607ccd3c0bc98295ab2ce84a77d322d/lib/addons/secrets-store/csi-driver-provider-aws-secrets.ts#L164

Reply:
addressed.